### PR TITLE
Refactor coord syntax to a tuple and refactor isBlack to color string

### PR DESCRIPTION
**Pull request by alvin-2002**

### CODE REVIEW

File: src/api/load_board.ts [hunk 1]
```diff
@@ -10,7 +10,8 @@ function convertNumToPiece(num: number): (Piece | null) {
     if(num === 0) return null;
 
     const piece: Piece = {
-        isBlack: (num & 0x8) === 8,
+        // (num & 0x8)=8 (truthy) then its black, if (num & 0x8)=0 (falsy) then its white
+        color: (num & 0x8) ? 'black' : 'white',
```
Comment (mitchell-merry) [inline src/api/load_board.ts:14]: I like this

File: src/api/moves.ts [hunk 2]
```diff
@@ -11,72 +11,71 @@ const moveGenerators: Record<PieceType, MoveGenerator> = {
...
 } 
 
-export function getValidMoves(board: Board, coord: Coordinate): Coordinate[] {
-    const currentPiece = board[coord.row][coord.col];
+export function getValidMoves(board: Board, [cellRow, cellCol]: Coordinate): Coordinate[] {
+    const currentPiece = board[cellRow][cellCol];
     if (currentPiece === null) return [];
 
-    return moveGenerators[currentPiece.type](board, coord);
+    return moveGenerators[currentPiece.type](board, [cellRow, cellCol]);
```
Comment (mitchell-merry) [inline src/api/moves.ts]: here there's no need to reconstruct the obj. would do:
```ts
export function getValidMoves(board: Board, coord: Coordinate): Coordinate[] {
    const [ row, col ] = coord;
    const currentPiece = board[row][col];
    ...
    return ...(board, coord);
    
}
```

File: src/api/moves.ts [hunk 3]
```diff
@@ -11,72 +11,71 @@ const moveGenerators: Record<PieceType, MoveGenerator> = {
...
 }
 
-export function coordOOB(coord: Coordinate): boolean {
-    return coord.row < 0 || coord.col < 0 || coord.row > 7 || coord.col > 7;
+export function coordOOB([cellRow, cellCol]: Coordinate): boolean {
```
Comment (mitchell-merry) [inline src/api/moves.ts]: good to destruct here but row, col is fine

File: src/assets/chesspieces.tsx [hunk 4]
```diff
@@ -18,7 +18,7 @@ import king_b from './images/king_b.png'
 
 import { PieceType } from '../api'
 
-export const chesspieces: Record<PieceType, any[]> = {
+export const chesspieces: Record<PieceType, string[]> = {
```
Comment (mitchell-merry) [inline src/assets/chesspieces.tsx:21]: good change. since create-react-app resolves these to strings on compilation

**Review by mitchell-merry — CHANGES_REQUESTED** [review]
change what was mentioned in disc and these comments and we should be good to go :D

File: src/api/moves.ts (hunk 2 above)
Comment (alvin-2002) [reply src/api/moves.ts]: Do you not want me to destructure the coordinate in this function parameter?

**Review by mitchell-merry — APPROVED** [review]
LGTM!

**Pull request merged**